noxfile-template.py

```diff
@@ -116,6 +177,52 @@ def py(session):
         print("SKIPPED: {} tests are disabled for this sample.".format(session.python))
 
 
+#
+# cloud_run session
+#
+
+@nox.session
+def cloud_run(session):
+    """Run tests for cloud run."""
+    if 'cloud_run' not in TEST_CONFIG['opt_in_sessions']:
+        print('SKIPPED: cloud_run tests are disabled for this sample.')
+        return
+
+    if os.path.exists("requirements.txt"):
+        session.install("-r", "requirements.txt")
+
+    if os.path.exists("requirements-test.txt"):
+        session.install("-r", "requirements-test.txt")
+
+    user_envs = get_pytest_env_vars()
+
+    # Only update gcloud on Kokoro.
+    if os.environ.get("KOKORO_JOB_NAME"):
+        # Update gcloud
+        session.run("gcloud", "components", "update", "--quiet")
+
+        # Activate service account
+        key_file = os.environ['GOOGLE_APPLICATION_CREDENTIALS']
+        session.run(
+            "gcloud", "auth", "activate-service-account",
+            "--key-file={}".format(key_file)
+        )
+
+        # Set gcloud project
+        project = user_envs['GCLOUD_PROJECT']
+        session.run("gcloud", "config", "set", "project", project)
```

Comment:
I don't think we should be updating gcloud in a test session - this could lead to non-deterministic test results, because now whenever a Cloud Run session is run gcloud could be at a different version for the tests after that. 

If we need a more updated version we should handle that either in the test script or on docker build.